### PR TITLE
Budget modify endpoints

### PR DESCRIPTION
**Pull request by fredberto**
#333

### CODE REVIEW

File: test/tests/budgeting/project_budget/modify.rs [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        .amount(new_project_budget_amount)
+        .send()
+        .await;
+
+    assert!(request.is_err());
+    assert_eq!(
+        request.unwrap_err().to_string(),
+        "Resource not found".to_string()
+    );
+}
```
Comment (gierens) [inline test/tests/budgeting/project_budget/modify.rs:277]: Would also be good to test that neither user nor master user can force an update.

**Review by gierens — CHANGES_REQUESTED** [review]
Looks mostly fine to me! Thanks! ... Only two things, there are a few corner cases that would be need to have tested, e.g. user and master user cannot force an update, ... and second and more importantly, please reuse the AuthorizationError variant instead of creating the new MethodNotAllowed variant as they are essentially identical and having both unnecessarily complicates things and is inconsistent with the rest of the api.
